lib/widgets/set_status.dart

```diff
@@ -0,0 +1,339 @@
+            bottom: 0),
+          child: Row(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: [
+              IconButton(
+                onPressed: chooseStatusEmoji,
+                style: IconButton.styleFrom(
+                  splashFactory: NoSplash.splashFactory,
+                  foregroundColor: designVariables.icon,
+                  tapTargetSize: MaterialTapTargetSize.shrinkWrap,
+                  padding: EdgeInsets.symmetric(
+                    vertical: 8,
+                    // In Figma design, there is no horizontal padding, but we
+                    // provide it in order to create a proper tap target size.
+                    horizontal: 8)),
+                icon: Row(children: [
+                  ValueListenableBuilder(
+                    valueListenable: statusChange,
+                    builder: (_, change, _) {
+                      final emoji = change.emoji.or(oldStatus.emoji);
+                      return emoji == null
+                        ? const Icon(ZulipIcons.smile, size: 24)
+                        : UserStatusEmoji(emoji: emoji, size: 24, neverAnimate: false);
+                    }),
+                  SizedBox(width: 4),
+                  Icon(ZulipIcons.chevron_down, size: 16),
+                ]),
+              ),
+              Expanded(child: TextField(
+                controller: statusTextController,
+                minLines: 1,
+                maxLines: 2,
+                maxLength: 60,
+                cursorColor: designVariables.textInput,
+                textCapitalization: TextCapitalization.sentences,
+                style: TextStyle(fontSize: 19, height: 24 / 19),
+                decoration: InputDecoration(
+                  counterText: '', // TODO: should we show counter?
+                  hintText: localizations.statusTextHint,
+                  hintStyle: TextStyle(color: designVariables.labelSearchPrompt),
+                  isDense: true,
+                  contentPadding: EdgeInsets.symmetric(
+                    vertical: 8,
+                    // Subtracting 4 pixels to account for the internal
+                    // 4-pixel horizontal padding.
+                    horizontal: 10 - 4,
+                  ),
+                  filled: true,
+                  fillColor: designVariables.bgSearchInput,
+                  border: OutlineInputBorder(
+                    borderRadius: BorderRadius.circular(10),
+                    borderSide: BorderSide.none,
+                  )))),
+            ]),
+        ),
+        Expanded(child: InsetShadowBox(
+          top: 6, bottom: 6,
+          color: designVariables.mainBackground,
+          child: SingleChildScrollView(
+            physics: AlwaysScrollableScrollPhysics(), // TODO: necessary?
+            padding: EdgeInsets.symmetric(vertical: 6),
+            child: Column(children: [
+              for (final status in statusSuggestions(context))
```

Comment:
Let's pull this out as a local variable: `final suggestions = statusSuggestions(context);`, and then this iterates through that local.

That way the call to this helper method is a bit more visible when reading this build method's logic.